share/server/loop.js

```diff
@@ -133,6 +133,9 @@ var Loop = function() {
     } else if (e.error && e.reason) {
       // compatibility with old error format
       respond(["error", e.error, e.reason]);
+    } else if (e.name == "InternalError") {
+      respond(["error", e.name, e.message]);
+      quit(-1);
```

Comment:
I think you probably want to move this conditional up to the second if the if/else chain or you run the risk of this being caught on line 133.

Reply:
Also, is this belt and suspended bracers? Becauses the change in view.js makes me think this should already be caught be the first conditional checking for "fatal".

Reply:
Good call on moving it up. It's belt and suspenders if we crash outside of the view function itself for instance when serializing the data to be returned back to the user.